Common/Securities/FutureOption/FuturesOptionsUnderlyingMapper.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            var futureChain = _chainProvider.GetFutureContractList(canonicalFutureSymbol, lookupDate);
+
+            foreach (var future in futureChain.OrderBy(s => s.ID.Date))
+            {
+                // Normalize by date first, normalize to a contract month date, then we want to get the contract
+                // month of the Future contract so we normalize by getting the delta between the expiration
+                // and the contract month.
+                var futureContractMonth = future.ID.Date.Date
+                    .AddDays(-future.ID.Date.Day + 1)
+                    .AddMonths(FuturesExpiryUtilityFunctions.GetDeltaBetweenContractMonthAndContractExpiry(future.ID.Symbol, future.ID.Date));
+
+                // We want a contract that is either the same as the contract month or greater
+                if (futureContractMonth < futureOptionContractMonth)
+                {
+                    continue;
+                }
+
+                return futureContractMonth;
+            }
+
+            // No matching/closest contract was found in the futures chain.
+            return null;
+        }
+
+        /// <summary>
+        /// Searches for the closest future's contract month depending on whether the Future Option's contract month is
+        /// on an even or odd month.
+        /// </summary>
+        /// <param name="futureOptionContractMonth">Future option contract month. Note that this is not the expiry of the Future Option.</param>
+        /// <param name="oddMonths">True if the Future Option's underlying future contract month is on odd months, false if on even months</param>
+        /// <returns>The underlying Future's contract month</returns>
+        protected DateTime ContractMonthEvenOddMonth(DateTime futureOptionContractMonth, bool oddMonths)
+        {
+            var monthEven = futureOptionContractMonth.Month % 2 == 0;
+            if (oddMonths && monthEven)
+            {
+                return futureOptionContractMonth.AddMonths(1);
+            }
+            if (!oddMonths && !monthEven)
+            {
+                return futureOptionContractMonth.AddMonths(1);
+            }
+
+            return futureOptionContractMonth;
+        }
+
+        /// <summary>
+        /// Sets the contract month to the third month for the first 3 months, then begins using the <see cref="ContractMonthEvenOddMonth"/> rule.
+        /// </summary>
+        /// <param name="futureOptionContractMonth">Future option contract month. Note that this is not the expiry of the Future Option.</param>
+        /// <param name="oddMonths">True if the Future Option's underlying future contract month is on odd months, false if on even months. Only used for months greater than 3 months</param>
+        /// <returns></returns>
+        protected DateTime ContractMonthYearStartThreeMonthsThenEvenOddMonthsSkipRule(DateTime futureOptionContractMonth, bool oddMonths)
```

Comment:
Believe should make these methods directly `private` since there doesn't seem to be a reason for them to be `protected`?